src/genie/libs/parser/asa/show_crypto_ikev2_sa.py

```diff
@@ -0,0 +1,221 @@
+            if m:
+                group = m.groupdict()
+                session_id = int(group['session_id'])
+                status = group['status']
+                ike_count = int(group['ike_count'])
+                child_sa_count = int(group['child_count'])
+                dict_session = dict_sessions.setdefault(session_id, {})
+                dict_session.update({'status': status})
+                dict_session.update({'ike_count': ike_count})
+                dict_session.update({'child_sa_count': child_sa_count})
+                dict_tunnels = dict_session.setdefault('tunnels', {})
+                continue
+
+            # 3752379 2001:db8:2:1::1/500 2001:db8:2:1::2/500 READY INITIATOR
+            m = p2.match(line)
+            if m:
+                group = m.groupdict()
+                tunnel_id = int(group['tunnel_id'])
+                local = group['local']
+                remote = group['remote']
+                status = group['status']
+                role = group['role']
+                dict_tunnel = dict_tunnels.setdefault(
+                    '{}-{}'.format(local, remote), {}
+                )
+                dict_tunnel.update({'tunnel_id': tunnel_id})
+                dict_tunnel.update({'local': local})
+                dict_tunnel.update({'remote': remote})
+                dict_tunnel.update({'status': status})
+                dict_tunnel.update({'role': role})
+                continue
+
+            # Encr: 3DES, Hash: SHA96, DH Grp:2, Auth sign: PSK, Auth verify: PSK
+            # Encr: AES-CBC, keysize: 256, Hash: SHA512, DH Grp:19, Auth sign: PSK, Auth verify: PSK
+            m = p3.match(line)
+            if m:
+                group = m.groupdict()
+                encr_alg = group['encr']
+                hash_alg = group['hash']
+                dh_group = int(group['dh_group'])
+                auth_sign = group['auth_sign']
+                auth_verify = group['auth_verify']
+                dict_tunnel.update({'encryption_algorithm': encr_alg})
+                dict_tunnel.update({'hashing_algorithm': hash_alg})
+                dict_tunnel.update({'dh_group': dh_group})
+                dict_tunnel.update({'authentication_sign': auth_sign})
+                dict_tunnel.update({'authentication_verify': auth_verify})
+                if 'keysize' in group and group['keysize']:
+                    dict_tunnel.update({
+                        'key_size': int(group['keysize'])
+                    })
+                continue
+
+            # Life/Active Time: 43200/53 sec
+            m = p4.match(line)
+            if m:
+                group = m.groupdict()
+                life_time = int(group['life_time'])
+                active_time = int(group['active_time'])
+                dict_tunnel.update({'lifetime_secs': life_time})
+                dict_tunnel.update({'activetime_secs': active_time})
+                child_sas = dict_tunnel.setdefault('child_sa', {})
```

Comment:
Probably safest to move this line down into p5 match section
```
# Child sa: local selector  2001:db8:1:1::/0 -
# 2001:db8:1:1:ffff:ffff:ffff:ffff/65535
m = p5.match(line)
if m:
    group = m.groupdict()
    local_selector = group['local_selector']
    child_sas = dict_tunnel.setdefault('child_sa', {})
    child_sa = child_sas.setdefault(child_sa_index, {})
    child_sa.update({'local_selector': str(local_selector)})
    child_sa_index = child_sa_index + 1
    continue
```
Just in case `Life/Active Time: X/X sec` doesn't appear in device output